### DISCUSSION TITLE
Sharing Playwright/Puppeteer instance?

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [avajs/ava#2748] [Q&A] [answered]
Is there a straightforward approach to this through shared workers?

**Comment by novemberborn** [comment, marked as answer]
I assume you'll launch the instance in the shared worker.

Could you elaborate on what test workers need to do with this shared instance?

**Reply by jakobrosenberg** [reply]
I'll confess that I'm completely out of my element when it comes to shared workers. What I'd like to do is have a number of test files all share the same Playwright browser instance.

I got the impression that it's doable from the quote below, but I have no idea if it's a straightforward task.

> Not sure what to build? Previously folks have expressed a desire for mutexes, managing Puppeteer instances, starting (database) servers and so forth.

https://github.com/avajs/ava/blob/97525a97c0f1e1fc609c980f6a4e66758c26480a/docs/recipes/shared-workers.md

**Reply by novemberborn** [reply]
There's been some discussion here: https://github.com/avajs/ava/discussions/2703#discussioncomment-567232

The problem I think is that you need some way to do remote procedure calls from the test worker to the shared Puppeteer instance. You can't share the full Puppeteer API directly.

**Reply by jakobrosenberg** [reply]
> You can't share the full Puppeteer API directly.

I feared you might say that. 😅
